research/fedora-packages-onboarding/onboard_package.py

```diff
@@ -0,0 +1,357 @@
+#!/usr/bin/env python3
+
+# Copyright Contributors to the Packit project.
+# SPDX-License-Identifier: MIT
+
+import logging
+import os
+import subprocess
+from pathlib import Path
+
+import click
+import git
+from yaml import safe_dump, safe_load
+
+from ogr import PagureService
+
+logger = logging.getLogger(__name__)
+logging.basicConfig(level=logging.INFO)
+
+RAWHIDE = "rawhide"
+ONBOARD_BRANCH_NAME = "packit-config"
+
+CONFIG_HEADER = """# See the documentation for more information:
+# https://packit.dev/docs/configuration/
+"""
+
+CONFIG_FILE_NAME = ".packit.yaml"
+PR_TITLE = """Add Packit configuration for automating release syncing"""
+PR_DESCRIPTION = """
+For more details, see https://packit.dev/docs/configuration/ or contact
+[the Packit team](https://packit.dev#contacts)
+
+"""
+
+
+@click.group()
+def cli() -> None:
+    pass
+
+
+def parse_actions_from_file(actions_file):
+    with open(actions_file) as file:
+        actions_content = file.read()
+
+    try:
+        actions = safe_load(actions_content)
+        if not isinstance(actions, dict):
+            raise ValueError("The content of the actions file is not a dictionary.")
+    except Exception as e:
+        raise ValueError(f"Error parsing YAML content: {e}")
+
+
+def generate_config_from_input(
+    upstream_project_url,
+    upstream_tag_template,
+    issue_repository,
+    upstream_tag_include,
+    upstream_tag_exclude,
+    include_koji_build,
+    include_bodhi_update,
+    allowed_committers,
+    allowed_pr_authors,
+    actions,
+):
+    config = {"upstream_project_url": upstream_project_url}
+    optional_config_keys = [
+        "upstream_tag_template",
+        "upstream_tag_include",
+        "upstream_tag_exclude",
+        "issue_repository",
+        "allowed_committers",
+        "allowed_pr_authors",
+        "actions",
+    ]
+
+    for key in optional_config_keys:
+        value = locals().get(key)
+        if value:
+            config[key] = value
+
+    config["jobs"] = [
+        {
+            "job": "pull_from_upstream",
+            "trigger": "release",
+            # TODO also make this as input parameter?
+            "dist_git_branches": ["fedora-all"],
```

Comment:
I am not sure here if we got this from input, whether these should be 3 separate arguments then (for each job), or one argument (pull_from_upstream and koji_build would use the input value directly and for bodhi-update we would compute the {targets-from-input} - {rawhide}

Reply:
thanks for the feedback, agreed!